### ISSUE TITLE
Support json

### ISSUE THREAD

**Issue by extrawurst** [RainwayApp/bebop#151]
**Is your feature request related to a problem? Please describe.**
I would really like to switch from proto to bebop but one thing it misses is supporting serialization to and from json for debugging.

**Describe the solution you'd like**
It would be great if bebop would add support to generate and decode to and from json aswell

**Comment by Eadword** [comment]
I see a lot of the projects you are in Rust. It would be pretty easy for me to add `serde` derive statements behind a feature flag which would allow using `serde_json` as well as the bebop serialization pathway.

**Comment by extrawurst** [comment]
> I see a lot of the projects on your home page are in Rust. It would be pretty easy for me to add `serde` derive statements behind a feature flag which would allow using `serde_json` as well as the bebop serialization pathway.

a feature flag would be perfect!

**Comment by Eadword** [comment]
We discussed this internally and at least for now we will add `serde` behind a flag for the Rust implementation, but we do not plan on implementing for other languages as it is non-trivial and out-of-scope.

With that said, I am going update this issue to track the `serde` support and not JSON support more widely. If you still want the full scope of the original ticket to be captured, please open a new ticket.

**Comment by Eadword** [comment]
This issue is blocked by #153

I ran into a challenge almost immediately with the `SliceWrapper` type and since there is already an open discussion around how to better handle lists and borrowing I am going to hold off on making a band-aid solution.
